### PR TITLE
ShrineOfTheStorm/Scripts: Implement Aqu'sirr encounter

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Implement Aqu'sirr encounter

**Tests performed:**

Tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

- [ ]  SPELL_EFFECT_CHARGE_DEST with trigger spell should trigger when reach destination

### CODE REVIEW

File: src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp [hunk 1]
```diff
@@ -0,0 +1,601 @@
...
+                DoCast(SPELL_GRASP_FROM_THE_DEPTHS_SELECTOR);
+                events.Repeat(38s);
+                break;
+            }
+            default:
+                break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
```
Comment (mdX7) [inline src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp:164]: you can drop UpdateAI entirely here, this behavior is default and ExecuteEvent is also called by default as you did it here)

File: src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+{
+    npc_aqusirr_intro_lord_stormsong(Creature* creature) : CreatureAI(creature) { }
+
+    void DoAction(int32 action) override
+    {
+        if (action == ACTION_START_INTRO)
+        {
+            me->RemoveAurasDueToSpell(SPELL_WATER_RITUAL);
+
+            _scheduler.Schedule(1s, [this](TaskContext context)
```
Comment (mdX7) [inline src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp:300]: dont try to build spaceships, use a variable `Milliseconds delay =1s;` and then use it as value for scheduling

File: src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp [hunk 3]
```diff
@@ -0,0 +1,601 @@
...
+            {
+                me->SetFacingTo(0.1939797f);
+
+                context.Schedule(3s, [this](TaskContext context)
+                {
+                    Talk(0);
```
Comment (mdX7) [inline src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp:306]: magic number

File: src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp [hunk 4]
```diff
@@ -0,0 +1,601 @@
...
+                    context.Schedule(6s, [this](TaskContext context)
+                    {
+                        Talk(1);
```
Comment (mdX7) [inline src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp:309]: magic number

File: src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp [hunk 5]
```diff
@@ -0,0 +1,601 @@
...
+};
+
+// 264911 - Erupting Waters
+// 264912 - Erupting Waters
+// 264913 - Erupting Waters
+class spell_aqusirr_erupting_waters_split : public SpellScript
+{
+    void SetDest(SpellDestination& dest)
+    {
+        switch (GetSpellInfo()->Id)
```
Comment (mdX7) [inline src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp:395]: move to separate scripts

File: src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp [hunk 6]
```diff
@@ -0,0 +1,601 @@
...
+    }
+};
+
+// 264941 - Erupting Waters
+class spell_aqusirr_erupting_waters : public SpellScript
+{
+    void HandleDamage(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        caster->SetHealth(caster->GetHealth() - caster->CountPctFromMaxHealth(GetEffectValue()));
```
Comment (mdX7) [inline src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp:499]: are you sure there is no spell for dealing the damage?

File: src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp [hunk 7]
```diff
@@ -0,0 +1,601 @@
...
+struct at_aqusirr_surging_rush : AreaTriggerAI
+{
+    at_aqusirr_surging_rush(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (!unit->IsPlayer())
+            return;
+
+        at->GetCaster()->CastSpell(unit, SPELL_SURGING_RUSH_DAMAGE, false);
```
Comment (mdX7) [inline src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp:579]: possible crash, caster may be null

**Review by mdX7 — CHANGES_REQUESTED** [review]

File: src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp (hunk 2 above)
Comment (aquadeus) [reply src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp:300]: i like spaceships :(

File: src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp (hunk 6 above)
Comment (aquadeus) [reply src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp:499]: same stuff as lady waycrest

**Pull request merged**